### DISCUSSION TITLE
Set Up WinPython Environment in Git Shell

### DISCUSSION THREAD

**Discussion by RoyiAvital** [winpython/winpython#1036] [Q&A]
I need to use WinPython within my Git Shell.
I thought about running `env.bat` to define WinPython environment but it doesn't work.

Any chance to add a BASH script which is equivalent?
Any other way to set it within Git Shell?

**Comment by stonebig** [comment]
call env.bat shall have worked.... what do you mean per Git Shell , or BASH ?

**Reply by stonebig** [reply]
'git CMD' shortcut is %LOCALAPPDATA%\Programs\Git\git-cmd.exe --cd-to-home
... but it doesn't handover the line

In reverse, you can call env.bat after

**Reply by stonebig** [reply]
I see it adds a EXEPATH=%LOCALAPPDATA%\Programs\Git

**Reply by stonebig** [reply]
and path is starting per Path=%LOCALAPPDATA%\Git\cmd

**Reply by stonebig** [reply]
so maybe try to improve env.bat like this:

after this
````
if %ERRORLEVEL% NEQ 0 (
   set "PATH=%WINPYDIR%\Lib\site-packages\PyQt5;%WINPYDIR%\Lib\site-packages\PySide2;%WINPYDIR%\;%WINPYDIR%\DLLs;%WINPYDIR%\Scripts;%WINPYDIR%\..\t;%WINPYDIR%\..\t\mingw32\bin;%WINPYDIR%\..\t\R\bin\x64;%WINPYDIR%\..\t\Julia\bin;%WINPYDIR%\..\n;%PATH%;"
   cd .
)         
````

add this:
````
rem Begin/Git integration 
set "PATH=%LOCALAPPDATA%\Git\cmd;%PATH%;"
set EXEPATH=%LOCALAPPDATA%\Programs\Git\git-cmd.exe
rem End/Git integration 
````

what do you think ?

**Comment by stonebig** [comment]
maybe try to improve env.bat like this:

after this
````
if %ERRORLEVEL% NEQ 0 (
   set "PATH=%WINPYDIR%\Lib\site-packages\PyQt5;%WINPYDIR%\Lib\site-packages\PySide2;%WINPYDIR%\;%WINPYDIR%\DLLs;%WINPYDIR%\Scripts;%WINPYDIR%\..\t;%WINPYDIR%\..\t\mingw32\bin;%WINPYDIR%\..\t\R\bin\x64;%WINPYDIR%\..\t\Julia\bin;%WINPYDIR%\..\n;%PATH%;"
   cd .
)         
````

add this:
````
rem Begin/Git integration 
set "PATH=%LOCALAPPDATA%\Git\cmd;%PATH%;"
set EXEPATH=%LOCALAPPDATA%\Programs\Git\git-cmd.exe
rem End/Git integration 
````

**Reply by stonebig** [reply]
not super quick, but does work apparently 
![image](https://user-images.githubusercontent.com/4312421/141653523-62215dab-56bc-4a18-8a7e-2c2addc42fdf.png)

**Reply by RoyiAvital** [reply]
I am not sure I get what you mean.
But this is the issue I'm having:

https://sendvid.com/hyh51g3m

What happens is that the whole Git Shell environment is lost while I want just the GitShell environment to be updated with the system variables defined in WinPython environment.

**Reply by stonebig** [reply]
Hum , you truly need a bash  shell version of env.bat.

**Reply by RoyiAvital** [reply]
I see.

Any chance to have it?

**Reply by stonebig** [reply]
If there is a DOS to bash translator on internet, there is a chance.

**Reply by RoyiAvital** [reply]
May I open an issue for it?
Maybe someone knowledgeable will handle it?

**Reply by stonebig** [reply]
sure no kitten will be armed if you open an issue on github and/or stackoverflow

**Reply by RoyiAvital** [reply]
By the way, Where is `env.bat` code on the repository? So I can link it from the issue.

**Reply by RoyiAvital** [reply]
@stonebig , Could you address where the `env.bat` code is in the repository?

**Comment by RoyiAvital** [comment]
I will give it a try when I'm back near the computer.
But the issue is running a `bat` file caused the Git Shell looks like a CMD window.
Something doesn't work there.

**Reply by stonebig** [reply]
The git-cmd.exe doesn't give back control to the caller,  so you have do what it does yourself, the  two lines suggested.

**Comment by stonebig** [comment]
and so ?

**Reply by RoyiAvital** [reply]
Not relevant anymore. I use Windows Terminal. Probably the best terminal out there.
